### PR TITLE
Create endpoint for refreshing of paymentmethods

### PR DESCRIPTION
**Pull request by holkra**
Haven't created any test so far. Feedback welcome which test actually would make sense here as the effect of this method call highly depends on the plugin that is responsible for the payment methods. And the endpoint doesn't do anything else than calling the service method.

### CODE REVIEW

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java [hunk 1]
```diff
@@ -766,6 +766,30 @@ public PaymentMethodJson apply(final PaymentMethod input) {
     }
 
     @TimedResource
+    @POST
+    @Path("/{accountId:" + UUID_PATTERN + "}/" + PAYMENT_METHODS + "/refresh")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Refresh account payment methods", response = PaymentMethodJson.class, responseContainer = "List")
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java]: Wrong `response` (nothing is returned).

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java [hunk 2]
```diff
@@ -766,6 +766,30 @@ public PaymentMethodJson apply(final PaymentMethod input) {
...
+                                          @QueryParam(QUERY_PLUGIN_PROPERTY) final List<String> pluginPropertiesString,
+                                          @HeaderParam(HDR_CREATED_BY) final String createdBy,
+                                          @HeaderParam(HDR_REASON) final String reason,
+                                          @HeaderParam(HDR_COMMENT) final String comment,
+                                          @javax.ws.rs.core.Context final HttpServletRequest request) throws AccountApiException, PaymentApiException {
+        final Iterable<PluginProperty> pluginProperties = extractPluginProperties(pluginPropertiesString);
+        final CallContext callContext = context.createContext(createdBy, reason, comment, request);
+
+        final Account account = accountUserApi.getAccountById(UUID.fromString(accountId), callContext);
+        paymentApi.refreshPaymentMethods(account, pluginName, pluginProperties, callContext);
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java]: If `pluginName` is null, you need to call the other `refreshPaymentMethods` API.

**Comment by pierre** [comment]
When you are done, can you squash the commits?

> Feedback welcome which test actually would make sense here as the effect of this method call highly depends on the plugin that is responsible for the payment methods.

Yes, the [mock plugin](https://github.com/killbill/killbill/blob/master/payment/src/test/java/org/killbill/billing/payment/provider/MockPaymentProviderPlugin.java#L288-L291) used in tests won't do much during a refresh. I would just check the return code (and try with a different input parameters, e.g. valid plugin name, invalid plugin name and no plugin name).

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java (hunk 1 above)
Comment (holkra) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java]: Fixed.

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java (hunk 2 above)
Comment (holkra) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java]: Fixed.

**Pull request merged**